### ISSUE TITLE
Best way to create prefab instances along roads

### ISSUE THREAD

**Issue by henning101** [mapbox/mapbox-unity-sdk#1651]
I wrote a custom MeshModifier in order to generate prefab instances along roads. The reason I use a mesh modifier and not a game object modifier is that within the mesh modifier I have access to all line vertices stored in VectorFeatureUnity. For a game object modifier I can only get the "tile center", "first vertex" or "center of vertices" and in my case I need all coordinates of each line string.

I know that this isn't the intended use case for the MeshModifier class but it works quite well. The problem is that I have to maintain an external list to store generated prefab instances and implement my own routines to destroy them if the user moves. MeshData only lets me store "mesh data" like vertices, normals, uvs etc.

Is there a better way of doing this? I thought about using the GeoJSON data from the UnityTile in GameObjectModifier but I assume the coordinates are unsorted here? In MeshModifier they're already grouped by road segment.

**Comment by brnkhy** [comment]
Hey @henning101!
It's kinda hacky (maybe a design flaw of modifiers) but that's exactly what I do as well. So great to hear you found and did that!

I'm not sure if I understand the question though. You create a list of those objects and modify/destroy them depending on char position right?
Ah if you're asking if there's a more elegant way of doing it, I don't think so. I mean if you want to push it, you probably can create a custom layer visualizer or modifier stack but unfortunately they are somewhat hardcoded in recent versions of the SDK so it's a little hard to add custom ones.
In short, I personally would do it with mesh modifiers as well (maybe just create another scriptable object and pass data to that, to keep things tidy etc). I don't like to be the bad news person but, you probably ran into problems in junctions right? Road decorations are not an easy task :(

**Comment by henning101** [comment]
Hey @brnkhy! Thank you for your answer! Yes junctions are indeed the problem because they need special texturing and things overlap. I went for a different style altogether where I use a custom Mapbox terrain image which is then processed using shaders. Works really well!

I would still love to see a road creation system in the Mapbox Unity package in the future! I'd be glad to contribute code

**Comment by brnkhy** [comment]
@henning101 I did a lot of experiments with road data in the past, especially on decorations and road meshes (with complex junctions meshes). Biggest problem at the moment is the data actually. We are using road data optimized for 2d maps which doesn't work too well with 3d maps.
Hopefully if that changes in the future, I'll definitely get back to this. 
![screen_5504x2924_2018-06-26_22-54-43](https://user-images.githubusercontent.com/1465390/93792347-147b0580-fc3e-11ea-8afd-78e1fc772fe5.jpg)

**Comment by Ojayyy** [comment]
Hi @brnkhy! Having read many posts from you and knowing that you are an expert on Mapbox. I want to beautify my mapbox landscape and roads. I haven't found a way yet or I'm not able to develop something like that myself. Can you provide me with a demo of a modified mesh modifier? How did you do it for example in the picture above?

![image](https://user-images.githubusercontent.com/13174622/152293875-325e3153-4eb7-4c95-bace-9724e206f81f.png)

**Comment by brnkhy** [comment]
Hey @Ojayyy !
unfortunately decorations are difficult! that screenshot above for example was using a different/special road data to make it easier. but that being said, most of those difficulties are more for city environments with a lot of road segments and buildings. If your use case is more like that's shown in your screenshot, it might be easier.
Idea is to use custom Mesh and Gameobject modifiers (under features tab of the map settings) to instantiate prefabs along road lines. SDK v2.1.1 doesn't have it built in though so it would require quite a bit of coding and procedural generation.

**Comment by Ojayyy** [comment]
I use the SDK 2.0.0 If I understand it correctly, it should be easier to implement with this version? I have looked at the custom mesh and gameobject modifier settings. Is this the setup that you recommended? What exactly do you mean with custom mesh?

![image](https://user-images.githubusercontent.com/13174622/152340978-fbecf4ab-59cb-433e-a02f-aeef06ea6a98.png)

**Comment by bigbangkobe** [comment]
> 为

how to do that
